### DISCUSSION TITLE
setx of OPENEMS_INSTALL_PATH did not work for me in windows 10

### DISCUSSION THREAD

**Discussion by ringo458** [thliebig/openEMS-Project#31] [General]
Using anaconda 3.10.9 environment. In the _init_ file I inserted print statements and saw that the variable was not found in environment even after I set it by the command line and it returned success. I know you have to restart the cmd window. I tried restarting the command window in anaconda and restarting anaconda. I don't know why the fonts are messed up I just copied from VIM.

In the end I had to do this:

```
# -*- coding: utf-8 -*-
#
# Shortcut CSXCAD import

from __future__ import absolute_import

AppCSXCAD_BIN = 'AppCSXCAD'


import os

print(os.name)
print(os.environ)
os.add_dll_directory('c:\opt\openEMS')
#print(os.environ['OPENEMS_INSTALL_PATH'])
#if os.name == 'nt' and 'OPENEMS_INSTALL_PATH' in os.environ and os.path.exists(os.environ['OPENEMS_INSTALL_PATH']):
#    try:  # to use Python 3.8's DLL handling
#        os.add_dll_directory(os.environ['OPENEMS_INSTALL_PATH'])
#    except AttributeError:  # <3.8, use PATH
#        if os.environ['OPENEMS_INSTALL_PATH'] not in os.environ['PATH']:
#            os.environ['PATH'] += os.pathsep + os.environ['OPENEMS_INSTALL_PATH']
#    
#    AppCSXCAD_BIN = os.path.join(os.environ['OPENEMS_INSTALL_PATH'], 'AppCSXCAD')
AppCSXCAD_BIN = os.path.join('c:\opt\openEMS', 'AppCSXCAD')

from CSXCAD.CSXCAD import ContinuousStructure
```

**Comment by thliebig** [comment]
This is of course a hacky way of doing it. There are of course other ways of setting windows environment variables.
What I do not understand is why you were unable to set this variable. I tested this on different Win10 machines without any issues...
How exactly did you try to set this?
You should be able to do it just using the cmd.exe....

**Reply by thliebig** [reply]
Btw: have you tried to just do
```
import os
os.environ['OPENEMS_INSTALL_PATH'] = 'c:\opt\openEMS'
import openEMS
```
E.g. at the beginning of the scripts? Of course this will not set it permanently...

**Reply by ringo458** [reply]
I've pretty much moved on to the Octave environment.
